xds/internal/balancer/clusterresolver/clusterresolver.go

```diff
@@ -252,7 +252,6 @@ func (b *clusterResolverBalancer) updateChildConfig() {
 	for i, a := range addrs {
 		endpoints[i].Attributes = a.BalancerAttributes
 		endpoints[i].Addresses = []resolver.Address{a}
-		endpoints[i].Addresses[0].BalancerAttributes = nil
```

Comment:
What was this here for before, and why is it removed?

Reply:
(I didn't write this code), but this was here before to clear the balancer attributes because we assumed this to be working.

This is needed for the way we do load reporting (temporary measure). A load report test was failing because the load reporting attribute wasn't stuck in the address. The old pf policy flattens the addresses from all the endpoints if endpoints are set and uses that (which it wasn't receiving prior to this change, so just uses addresses), but the issue is this was clearing the load reporting. Thus, the cluster impl policy wasn't looking at the top level endpoint load reporting, but this. This is all temporary code that will be phased out but that is the technical reason.

Reply:
OK that makes sense, thanks.